modules/effects/arrow-effects-kotlinx-coroutines/src/main/kotlin/arrow/effects/DeferredK.kt

```diff
@@ -112,9 +113,32 @@ data class DeferredK<out A>(private val deferred: Deferred<A>, val scope: Corout
           result
         }.k()
       }
+
+    fun <A, B> racePair(ctx: CoroutineContext,
+                        lh: DeferredKOf<A>,
+                        rh: DeferredKOf<B>): DeferredK<Either<Tuple2<A, Fiber<ForDeferredK, B>>, Tuple2<Fiber<ForDeferredK, A>, B>>> =
+      lh.startF(ctx).flatMap { fiberA ->
+        rh.startF(ctx).flatMap { fiberB ->
+          DeferredK.async<Either<Tuple2<A, Fiber<ForDeferredK, B>>, Tuple2<Fiber<ForDeferredK, A>, B>>> { cb ->
+            fiberA.join.fix().invokeOnCompletion { error ->
```

Comment:
Use use normal operators like `unsafeRunAsync` for this case.

If that's not available I made a new function that's sometimes required to trigger `invokeOnCompletion` for Deferred. Check `forcePropagateExceptions` or something like that.